src/blockchain/btc_blockchain_util.cpp

```diff
@@ -99,10 +99,49 @@ uint32_t getNextWorkRequired(const BlockIndex<BtcBlock>& prevBlock,
   uint32_t nHeightFirst =
       prevBlock.height - (params.getDifficultyAdjustmentInterval() - 1);
   const auto* pindexFirst = prevBlock.getAncestor(nHeightFirst);
-  assert(pindexFirst);
+  if (pindexFirst == nullptr) {
+    throw std::logic_error("unable to find block ancestor at given height");
+  }
 
   return calculateNextWorkRequired(
       prevBlock, pindexFirst->getBlockTime(), params);
 }
 
+template <>
+int64_t getMedianTimePast(const BlockIndex<BtcBlock>& prev) {
+  static constexpr int medianTimeSpan = 11;
+
+  int64_t pmedian[medianTimeSpan];
+  std::fill(pmedian, pmedian + medianTimeSpan, 0);
+  auto* pbegin = &pmedian[medianTimeSpan];
```

Comment:
If prev BlockIndex is null pbegin will point to pmedian.end(). Therefore any access to pbegin will throw an exception.

Reply:
how prev can be null? it is passed as reference in this function